.gitattributes

```diff
@@ -0,0 +1,2 @@
+* text=auto
+*.cs text eol=crlf
```

Comment:
Because of this: https://github.com/OrchardCMS/OrchardCore.Commerce/runs/6181940340?check_suite_focus=true
Looks like the IDE analyzer doesn't work well with LF. I have verified this on Linux too.
Indeed I will have to apply this to OSOCE before OSOCE-49 can be released. Just a bit later... I am tired now.

Reply:
I remember that Zoltán has already chosen to disable some of those IDE analyzers because they produce more noise than value.

Have you tried publishing one of the OSOCE submodule NuGet packages, e.g. Privacy, after the switch to `* text=auto`? Shouldn't that build throw the same kind of error? Also, we run the OSOCE build in TC and it doesn't fail with that warning. How is that possible?

Reply:
> I remember that Zoltán has already chosen to disable some of those IDE analyzers because they produce more noise than value.

This is not among them. Normally IDE0055 is pretty useful, but if it identifies newlines as CRLF, then it will think the file is collapsed into a single line when only LF are used.


> Also, we run the OSOCE build in TC and it doesn't fail with that warning. How is that possible?

TC is running Windows. The "text=auto" means the files are checked out with native line endings and committed with LF. So on Windows files are checked out CRLF. In the GitHub action I have linked the Windows version runs successfully too, there is no problem with that side. On Linux the same files are checked out with LF, so if an analyzer requires CRLF then it will misbehave. It seems like we have to stick with CRLF for .cs files. This kind of granularity is not unusual, it's just a shame it's necessary. I don't know if this is a bug or an intentional decision in the analyzer.

Reply:
It seems like an annoyance that should be fixed in the analyzer. We might choose to tackle that in a separate issue. Agreed?

Reply:
I wonder if [this](https://github.com/dotnet/roslyn/issues/61042) is related.

Reply:
Possible, I have subscribed to that issue and will keep an eye on it.